### ISSUE TITLE
Remote debug & live edit doesn't work on 2.0 alpha

### ISSUE THREAD

**Issue by volzhs** [godotengine/godot#2401]
http://www.godotengine.org/forum/viewtopic.php?f=12&t=1523

is this issue resolved?

**Comment by okamstudio** [comment]
Yup

**Comment by volzhs** [comment]
I tried to debug on android devices.
So, I checked on "Deploy Remote Debug" option and run on android.

and adb rogcat says.

```
08-28 23:16:40.565: I/godot(8274): [1;31;40mERROR: _resolve_hostname: [1;37;40mgethostbyname failed!
08-28 23:16:40.565: I/godot(8274): [0;31;40m   At: drivers/unix/ip_unix.cpp:69.[0;0;37m
08-28 23:16:40.566: I/godot(8274): [1;31;40mERROR: connect: [1;37;40mCondition ' p_host.host == 0 ' is true. returned: ERR_INVALID_PARAMETER
08-28 23:16:40.566: I/godot(8274): [0;31;40m   At: drivers/unix/stream_peer_tcp_posix.cpp:138.[0;0;37m
08-28 23:16:40.887: W/ActivityManager(511): Activity pause timeout for ActivityRecord{de61d2 u0 com.macaron.twinpuzzle/com.android.godot.Godot t3378}
08-28 23:16:40.921: E/cpu_temp(6690): MSG_ON_WATCH_CURRENT
08-28 23:16:41.200: D/WifiService(511): New client listening to asynchronous messages
08-28 23:16:41.566: I/godot(8274): Remote Debugger: Connection failed with status: 0'', retrying in 1 sec.
08-28 23:16:42.567: I/godot(8274): Remote Debugger: Connection failed with status: 0'', retrying in 1 sec.
08-28 23:16:43.568: I/godot(8274): Remote Debugger: Connection failed with status: 0'', retrying in 1 sec.
08-28 23:16:43.568: I/godot(8274): Remote Debugger: Unable to connect
08-28 23:16:43.568: I/godot(8274): pre-removing print handler...
08-28 23:16:43.568: I/godot(8274): removing print handler...
08-28 23:16:43.568: I/godot(8274): found
08-28 23:16:43.568: I/godot(8274): print hanlder list is 0x0
08-28 23:16:43.577: I/godot(8274): has res dir: 
08-28 23:16:43.579: I/godot(8274): exists engine cfg? 0
08-28 23:16:43.583: I/godot(8274): loaded engine.cfg
08-28 23:16:43.588: I/godot(8274): *****SETUP OK
08-28 23:16:43.654: I/godot(8274): **START
```

What should I do to debug on android?
I use c2e2f2e0aebf6342e6f18ae5d67b6a825590675a commit (Aug 8, 2015) for now.

**Comment by okamstudio** [comment]
check a few things, mainly that

1) your export templates are as recent as your editor
2) both your device and your PC are in the same intranet and can see
eachother (remote debug goes via wifi, not via usb cable)

if both are met, it should work



On Fri, Aug 28, 2015 at 11:21 AM, volzhs <notifications@github.com> wrote:

> I tried to debug on android devices.
> So, I checked on "Deploy Remote Debug" option and run on android.
>
> and adb rogcat says.
>
> 08-28 23:16:40.565: I/godot(8274): �[1;31;40mERROR: _resolve_hostname: �[1;37;40mgethostbyname failed!
> 08-28 23:16:40.565: I/godot(8274): �[0;31;40m   At: drivers/unix/ip_unix.cpp:69.�[0;0;37m
> 08-28 23:16:40.566: I/godot(8274): �[1;31;40mERROR: connect: �[1;37;40mCondition ' p_host.host == 0 ' is true. returned: ERR_INVALID_PARAMETER
> 08-28 23:16:40.566: I/godot(8274): �[0;31;40m   At: drivers/unix/stream_peer_tcp_posix.cpp:138.�[0;0;37m
> 08-28 23:16:40.887: W/ActivityManager(511): Activity pause timeout for ActivityRecord{de61d2 u0 com.macaron.twinpuzzle/com.android.godot.Godot t3378}
> 08-28 23:16:40.921: E/cpu_temp(6690): MSG_ON_WATCH_CURRENT
> 08-28 23:16:41.200: D/WifiService(511): New client listening to asynchronous messages
> 08-28 23:16:41.566: I/godot(8274): Remote Debugger: Connection failed with status: 0'', retrying in 1 sec.
> 08-28 23:16:42.567: I/godot(8274): Remote Debugger: Connection failed with status: 0'', retrying in 1 sec.
> 08-28 23:16:43.568: I/godot(8274): Remote Debugger: Connection failed with status: 0'', retrying in 1 sec.
> 08-28 23:16:43.568: I/godot(8274): Remote Debugger: Unable to connect
> 08-28 23:16:43.568: I/godot(8274): pre-removing print handler...
> 08-28 23:16:43.568: I/godot(8274): removing print handler...
> 08-28 23:16:43.568: I/godot(8274): found
> 08-28 23:16:43.568: I/godot(8274): print hanlder list is 0x0
> 08-28 23:16:43.577: I/godot(8274): has res dir:
> 08-28 23:16:43.579: I/godot(8274): exists engine cfg? 0
> 08-28 23:16:43.583: I/godot(8274): loaded engine.cfg
> 08-28 23:16:43.588: I/godot(8274): *****SETUP OK
> 08-28 23:16:43.654: I/godot(8274): **START
>
> What should I do to debug on android?
> I use c2e2f2e
> <https://github.com/okamstudio/godot/commit/c2e2f2e0aebf6342e6f18ae5d67b6a825590675a>
> commit (Aug 8, 2015) for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/2401#issuecomment-135786709>.
>



-- 

*OkamStudio*

**Comment by reduz** [comment]
I also just noticed that Android added adb reverse, this should eventually
allow to add a more efficient remotefs

**Comment by volzhs** [comment]
@okamstudio 
I recompiled tools and templates. (still I'm using c2e2f2e0aebf6342e6f18ae5d67b6a825590675a)
And connect device with ``adb tcpip 5555``

```
C:\Users\Administrator>adb devices
List of devices attached
192.168.0.12:5555       device
```

I updated tools and template, and connect over wifi in same intranet.
But logcat says same messages.

Should I use latest commit?

**Comment by volzhs** [comment]
I finally found what causes this problem.

I found empty string for ``network/debug_host`` in ``editor_setting.xml``.
```
<string name="network/debug_host"> "" </string>
```

but Settings > Editor Settings > Network > Debug host shows proper IP.
I never change this property before.

So, I changed it something another and change it back to default IP.
Then, ``editor_setting.xml`` is saved properly.

And debug remote & live edit seems to work now. :)
